src/jit/compiler.hpp

```diff
@@ -1456,20 +1456,38 @@ inline void GenTree::SetOper(genTreeOps oper, ValueNumberUpdate vnUpdate)
     }
 }
 
-inline void GenTree::CopyFrom(const GenTree* src, Compiler* comp)
+//------------------------------------------------------------------------
+// CopyFrom: replace this with the src node. If the source is isolated node, then use it as it.
+//           Otherwise, make a src clone.
+//
+// Arguments:
+//    src - source tree, that replaces this.
+//    comp - the compiler instanse to transfer annotations for arrays.
+//
+inline void GenTree::CopyFrom(GenTree* src, Compiler* comp)
 {
     /* The source may be big only if the target is also a big node */
 
     assert((gtDebugFlags & GTF_DEBUG_NODE_LARGE) || GenTree::s_gtNodeSizes[src->gtOper] == TREE_NODE_SZ_SMALL);
-    GenTreePtr prev = gtPrev;
-    GenTreePtr next = gtNext;
+
+    // To satisfy SDSU we can copy only isolated nodes, that are not in the blocks.
+    // If the source is not isolated - create a clone.
+    if ((src->gtNext != nullptr) || (src->gtPrev != nullptr))
+    {
+        src = comp->gtCloneExpr(src);
+    }
+
+    assert((src->gtNext == nullptr) && (src->gtNext == nullptr));
+
+    // Mark that after the copying source is built-in the blocks.
```

Comment:
I'd suggest rewording this as "Copy the execution order links to 'src', so that when we copy it to 'this', below, those links remain correct."